frappe/public/js/frappe/ui/toolbar/awesome_bar.js

```diff
@@ -324,29 +324,91 @@ frappe.search.AwesomeBar = class AwesomeBar {
 		}
 	}
 
-	make_calculator(txt) {
-		var first = txt.substr(0, 1);
-		if (first == parseInt(first) || first === "(" || first === "=") {
-			if (first === "=") {
-				txt = txt.substr(1);
-			}
-			try {
-				var val = eval(txt);
-				var formatted_value = __("{0} = {1}", [txt, (val + "").bold()]);
-				this.options.push({
-					label: formatted_value,
-					value: __("{0} = {1}", [txt, val]),
-					match: val,
-					index: 80,
-					default: "Calculator",
-					onclick: function () {
-						frappe.msgprint(formatted_value, __("Result"));
-					},
-				});
-			} catch (e) {
-				// pass
+	make_calculator(raw) {
+		const firstChar = raw.charAt(0);
+
+		// Remove everything but digits, word characters, or opening parentheses
+		// from the beginning as well as trailing whitespace and "=".
+		let expr = raw.replace(/^[^\d\w(-]+/i, '').replace(/[\s=]+$/g, '');
+
+		// Store the original text for later
+		const original = expr;
+
+		// Detect currency symbols and ISO 4217 currency codes
+		const currencyCodes = ['EUR', 'USD', 'GBP', 'JPY', 'INR'].join("|");
+		const currencyRegex = new RegExp(`(${currencyCodes}|\\p{Sc})`, 'gu');
+		const detectedCurrencies = expr.match(currencyRegex) || [];
+		let currency = "";
+		switch (detectedCurrencies.length) {
+			case 0:
+				break;
+			case 1:
+				expr = expr.replace(currencyRegex, "");
+				currency = detectedCurrencies[0];
+				break;
+			default:
+				// Multiple currencies are not supported.
+				return;
+		}
+
+		// Replace single quotes as a separator
+		expr = expr.replace(/'/g, '');
+
+		// Remove Indian lakh and crore separators
+		expr = expr.replace(/(\d{1,2})?,?(\d{1,2}),(\d{2},\d{3})\.(\d+)/g, '$1$2$3.$4');
+
+		// Remove commas, straight and curly apostrophes used as thousands separators
+		expr = expr.replace(/(\d)[\s'’_,\.](?=\d{3}([^\d]|$))/g, '$1');
+
+		// Replace European decimal commas by American decimal points
+		expr = expr.replace(/(\d+),(\d{1,2})($|[^\d])/g, '$1.$2$3');
+
+		// Replace '^' with Javascript's '**' for exponentiation
+		expr = expr.replace(/\^/g, '**');
+
+		// Remove superfluous spaces
+		expr = expr.replace(/ /g, '');
```

Comment:
Umm, none of this seems simple or maintainable, let's not complicate this? It's supposed to be a simple calculator for very basic arithmetic.

Reply:
It should be a simple to use, basic calculator, true. Under the hood it will always be quite complex even to get simple things right.

I was currently working on reimplementing Math functions, which have been supported before and probably shouldn‘t disappear, while hardening our potentially risky use of `eval()`. This will add even more code, and a full-scale tokenizer it would make it even more complex.

But maybe you‘re right. As a rather gimmicky by-feature this one primarily needs to be maintainable. With this in mind, the best way would be to look closer which functionality may be necessary to fix and improve more central parts of Frappe, for example the editable grid, and then reuse that functionality.

From the original issue:

> https://github.com/frappe/frappe/assets/710051/8d832af3-1bb5-42aa-9dab-316e8006b555
> 
> _Originally posted by @TurkerTunali in https://github.com/frappe/frappe/issues/22219#issuecomment-1715055201_

Reply:
Instead of currency calculations, simple 4 operator math calculations is sufficient.

Reply:
Ok, I see, this PR needs to be closed and either spun off to a contrib app, or left as it is.

Reply:
we are planning awesomebar plugin APIs or newer version of it. Then these can be added as plugins.